index.js

```diff
@@ -255,61 +291,72 @@ class CoverageSonar extends CoverageBase {
-            .then(() => createUser(username, password))
-            .then(() => grantUserPermission(username, projectKey))
-            .then(() => generateToken(username))
+        return createProject(projectData.projectKey)
+            .then(() => createUser(projectData.username, password))
+            .then(() => grantUserPermission(projectData.username, projectKey))
+            .then(() => generateToken(projectData.username))
             .then(res => res.token);
     }
 
     /**
      * Return links to the Sonar project and coverage metadata
      * @method getInfo
      * @param   {Object}  config
-     * @param   {Object}  [config.annotations]      Screwdriver job annotations
+     * @param   {Object}  [config.scope]            Coverage scope
      * @param   {String}  config.jobId              Screwdriver job ID
      * @param   {String}  config.jobName            Screwdriver job name
      * @param   {String}  config.pipelineId         Screwdriver pipeline ID (if enterprise is enabled)
      * @param   {String}  config.pipelineName       Screwdriver pipeline name
      * @param   {String}  [config.prNum]            Pull request number
+     * @param   {String}  [config.prParentJobId]    Pull request parent job ID
      * @param   {String}  config.startTime          Job start time
      * @param   {String}  config.endTime            Job end time
-     * @param   {String}  [config.coverageProjectKey]  Sonar project key
+     * @param   {String}  [config.projectKey]       Sonar project key
      * @return  {Promise}                           An object with:
      *                                              - tests success percentage
      *                                              - coverage percentage
      *                                              - project url
      *                                              - Sonar env vars
      */
-    _getInfo({ annotations, jobId, jobName, startTime, endTime, pipelineId,
-        pipelineName, prNum, coverageProjectKey }) {
-        const { projectKey: computedProjectKey, projectName } = getProjectData({
+    _getInfo({ scope, jobId, jobName, startTime, endTime, pipelineId,
+        pipelineName, prNum, projectKey: coverageProjectKey, prParentJobId }) {
+        const { projectKey, projectName, username } = getProjectData({
             enterpriseEnabled: sonarEnterprise,
             jobId,
             pipelineId,
-            annotations,
+            scope,
             pipelineName,
-            jobName
+            jobName,
+            projectKey: coverageProjectKey,
+            prParentJobId,
+            prNum
         });
-        const projectKey = coverageProjectKey || computedProjectKey;
         const infoObject = {
             envVars: {
-                SD_SONAR_AUTH_URL: sdCoverageAuthUrl,
+                // eslint-disable-next-line max-len
+                SD_SONAR_AUTH_URL: `${sdCoverageAuthUrl}?projectKey=${projectKey}&username=${username}`,
```

Comment:
`scope` also need to be added. 

Else it will do an unnecessary lookup https://github.com/screwdriver-cd/screwdriver/pull/2167/files#diff-63d46bdd1d4c5dd94f20824c664b4706R41

Reply:
added